src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -572,21 +602,57 @@ public void run() {
       log.info(String.format("NameNodes: %s", persistentState.getNameNodes()));
       log.info(String.format("DataNodes: %s", persistentState.getDataNodes()));
 
-      LinkedHashMap<Protos.TaskID, Protos.TaskStatus> runningTasks = liveState.getRunningTasks();
       Collection<String> taskIds = persistentState.getAllTaskIds();
-      Collection<Protos.TaskID> runningTaskIds = runningTasks.keySet();
+      Set<String> runningTaskIds = liveState.getRunningTasks().keySet();
       for (String taskId : taskIds) {
-        boolean taskFound = false;
-        for (Protos.TaskID runningTaskId : runningTaskIds) {
-          if (runningTaskId.getValue().equals(taskId)) {
-            taskFound = true;
-          }
-        }
-        if (!taskFound) {
+        if (!runningTaskIds.contains(taskId))
           persistentState.removeTaskId(taskId);
-        }
       }
       reconciliationCompleted = true;
     }
   }
+
+  private boolean journalNodesAvailable() {
+    Set<String> hosts = new HashSet<>();
+    if (conf.usingMesosDns()) {
+      for (int i = conf.getJournalNodeCount(); i > 0; i--)
```

Comment:
Why backwards? To fail earlier if JN-last is unavailable? Worth a comment so others don't naively "fix" your reverse iteration.